From 4225f1210c5773ebdd418d382b30c85ed57014d7 Mon Sep 17 00:00:00 2001
From: Mika Laitio <mika.laitio@amd.com>
Date: Sun, 23 Nov 2025 01:29:28 -0800
Subject: [PATCH 5/5] install newer cmake via pip

Signed-off-by: Mika Laitio <mika.laitio@amd.com>
---
 rcb_init.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rcb_init.sh b/rcb_init.sh
index 9e077273..3ef46d34 100755
--- a/rcb_init.sh
+++ b/rcb_init.sh
@@ -4,4 +4,5 @@ python3 -m venv ./.venv && source ./.venv/bin/activate
 pip install --upgrade pip
 pip install -r ./requirements.txt
 pip install "dvc[s3]"
+pip install cmake
 python ./build_tools/fetch_sources.py
-- 
2.43.0

