From a4616fa026d8814b53d3a9194ec6f3a686ea3a5d Mon Sep 17 00:00:00 2001
From: Mika Laitio <mika.laitio@amd.com>
Date: Mon, 16 Mar 2026 11:16:12 -0700
Subject: [PATCH 10/10] enable parallel compile and link options for llvm

Signed-off-by: Mika Laitio <mika.laitio@amd.com>
---
 rcb_config.py | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/rcb_config.py b/rcb_config.py
index 27deff5..52e2e4b 100644
--- a/rcb_config.py
+++ b/rcb_config.py
@@ -34,6 +34,8 @@ def main():
     print(f"rcb_config.py CMAKE_VERSION: {cmake_version}")
 
     amdgpu_targets = os.environ.get("RCB_AMDGPU_TARGETS", "")
+    safe_cpu_count_compile = os.environ.get("RCB_SAFE_CPU_COUNT_BUILD", "")
+    safe_cpu_count_link = os.environ.get("RCB_SAFE_CPU_COUNT_LINK", "")
     cmake_cmd = [
         "cmake",
         "-B", "build",
@@ -50,6 +52,15 @@ def main():
         if answer == "y":
             cmake_cmd.append("-DTHEROCK_SANITIZER=ASAN")
     cmake_cmd.append(f"-DTHEROCK_AMDGPU_FAMILIES={amdgpu_targets}")
+    print(f"safe_cpu_count_compile: {safe_cpu_count_compile} safe_cpu_count_link: {safe_cpu_count_link}")
+    if safe_cpu_count_compile != "":
+        cmake_cmd.append(f"-DLLVM_PARALLEL_COMPILE_JOBS={safe_cpu_count_compile}")
+        flang_safe_cpu_compile_count = int(safe_cpu_count_compile) - 2
+        flang_safe_cpu_compile_count = max(1, flang_safe_cpu_compile_count)
+        flang_safe_cpu_compile_count = str(flang_safe_cpu_compile_count)
+        cmake_cmd.append(f"-DFLANG_PARALLEL_COMPILE_JOBS={flang_safe_cpu_compile_count}")
+    if safe_cpu_count_link != "":
+        cmake_cmd.append(f"-DLLVM_PARALLEL_LINK_JOBS={safe_cpu_count_link}")
     if IS_WINDOWS:
         cmake_cmd.append("-DTHEROCK_AMDGPU_DIST_BUNDLE_NAME=windows")
     cmake_cmd.append(".")
-- 
2.43.0

