From d70906f275d83da811a34ccbeb1987a993d48535 Mon Sep 17 00:00:00 2001
From: Mika Laitio <mika.laitio@amd.com>
Date: Fri, 20 Mar 2026 14:32:09 -0700
Subject: [PATCH 13/16] update asan check in rcb_config.py

Signed-off-by: Mika Laitio <mika.laitio@amd.com>
---
 rcb_config.py | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/rcb_config.py b/rcb_config.py
index 27deff5..27bc359 100644
--- a/rcb_config.py
+++ b/rcb_config.py
@@ -44,11 +44,14 @@ def main():
     sanitizer = os.environ.get("RCB_THEROCK_SANITIZER", "")
     if sanitizer:
         cmake_cmd.append(f"-DTHEROCK_SANITIZER={sanitizer}")
-    elif amdgpu_targets in asan_gpu_families:
-        answer = input(f"RCB_THEROCK_SANITIZER is not set and target is {amdgpu_targets}. "
+    else:
+        for asan_gpu_item in asan_gpu_families:
+            if asan_gpu_item in amdgpu_targets:
+                answer = input(f"RCB_THEROCK_SANITIZER is not set and target is {amdgpu_targets}. "
                        "Enable ASAN build? [y/N]: ").strip().lower()
-        if answer == "y":
-            cmake_cmd.append("-DTHEROCK_SANITIZER=ASAN")
+                if answer == "y":
+                    cmake_cmd.append("-DTHEROCK_SANITIZER=ASAN")
+                    break
     cmake_cmd.append(f"-DTHEROCK_AMDGPU_FAMILIES={amdgpu_targets}")
     if IS_WINDOWS:
         cmake_cmd.append("-DTHEROCK_AMDGPU_DIST_BUNDLE_NAME=windows")
-- 
2.41.3

