From 1e485c3aec284277a603a82c2fbe88ea7a2402d5 Mon Sep 17 00:00:00 2001
From: Mika Laitio <mika.laitio@amd.com>
Date: Wed, 15 Apr 2026 20:07:50 -0700
Subject: [PATCH 16/23] rocprofiler-sdk do not build external yaml-cpp

Signed-off-by: Mika Laitio <mika.laitio@amd.com>
---
 profiler/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/profiler/CMakeLists.txt b/profiler/CMakeLists.txt
index 0a55b564..7328d8a2 100644
--- a/profiler/CMakeLists.txt
+++ b/profiler/CMakeLists.txt
@@ -102,6 +102,7 @@ if(THEROCK_ENABLE_ROCPROFV3)
       -DROCPROFILER_BUILD_INTEGRATION_TESTS=OFF
       -DROCPROFILER_BUILD_SAMPLES=OFF
       -DROCPROFILER_BUILD_ELFIO=OFF
+      -DROCPROFILER_BUILD_YAML_CPP=OFF
       -DCMAKE_PREFIX_PATH="${THEROCK_BINARY_DIR}/dist/rocm"
       -DROCM_PATH="${THEROCK_BINARY_DIR}/dist/rocm"
       -DROCPROFILER_PYTHON_VERSIONS:STRING="${_detected_python_versions_str}"
@@ -116,6 +117,7 @@ if(THEROCK_ENABLE_ROCPROFV3)
       amd-hip
     BUILD_DEPS
       therock-elfio
+      therock-yaml-cpp
     RUNTIME_DEPS
       aqlprofile
       hip-clr
-- 
2.52.0

