From 44322f2acbd2dfa5de54840b91f7597aecb2d8bd Mon Sep 17 00:00:00 2001
From: Mika Laitio <mika.laitio@amd.com>
Date: Tue, 12 May 2026 11:43:11 -0700
Subject: [PATCH 18/21] rocprofiler-systems use therock provided elfutils

Signed-off-by: Mika Laitio <mika.laitio@amd.com>
---
 profiler/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/profiler/CMakeLists.txt b/profiler/CMakeLists.txt
index f03dc60..495e1dc 100644
--- a/profiler/CMakeLists.txt
+++ b/profiler/CMakeLists.txt
@@ -324,6 +324,7 @@ if (THEROCK_ENABLE_ROCPROFSYS AND THEROCK_SANITIZER STREQUAL "")
       -DROCPROFSYS_GFX_TARGETS="${_ROCPROFSYS_GFX_TARGETS_ESCAPED}"
       -DROCPROFSYS_BUILD_SPDLOG=OFF
       -DROCPROFSYS_BUILD_SQLITE3=OFF
+      -DROCPROFSYS_BUILD_ELFUTILS=OFF # use elfutils provided by therock
       -DROCPROFSYS_USE_PYTHON="${_ROCPROFSYS_USE_PYTHON}"
       -DROCPROFSYS_PYTHON_VERSIONS:STRING="${_ROCPROFSYS_PYTHON_VERSIONS_STR}"
       -DROCPROFSYS_PYTHON_ROOT_DIRS:STRING="${_ROCPROFSYS_PYTHON_ROOT_DIRS_STR}"
@@ -336,6 +337,7 @@ if (THEROCK_ENABLE_ROCPROFSYS AND THEROCK_SANITIZER STREQUAL "")
     CMAKE_INCLUDES
       therock_explicit_finders.cmake
     BUILD_DEPS
+      therock-elfutils
       therock-spdlog
       therock-fmt
       roctracer
-- 
2.52.0

