From b1f340338319826a7c3cd875fbb2873a44694c26 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Thu, 30 Oct 2025 17:10:33 -0700
Subject: [PATCH] printout stacktrace on error case

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 c10/cuda/CUDAMiscFunctions.cpp | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/c10/cuda/CUDAMiscFunctions.cpp b/c10/cuda/CUDAMiscFunctions.cpp
index 95a424f..98f1713 100644
--- a/c10/cuda/CUDAMiscFunctions.cpp
+++ b/c10/cuda/CUDAMiscFunctions.cpp
@@ -2,8 +2,26 @@
 #include <c10/util/env.h>
 #include <string>
 
+#if defined(USE_ROCM)
+#include <c10/util/Backtrace.h>
+#include <iostream>
+#endif
+
 namespace c10::cuda {
 
+#if defined(USE_ROCM)
+namespace {
+
+void print_stack_trace() noexcept {
+  try {
+    std::cerr << c10::get_backtrace(1);
+  } catch (...) {
+  }
+}
+
+} // namespace
+#endif
+
 // Explain common CUDA errors
 // NOLINTNEXTLINE(bugprone-exception-escape,-warnings-as-errors)
 std::string get_cuda_error_help(cudaError_t error) noexcept {
@@ -14,6 +32,9 @@ std::string get_cuda_error_help(cudaError_t error) noexcept {
           "\nGPU device may be out of range, do you have enough GPUs?");
       break;
     default:
+#if defined(USE_ROCM)
+      print_stack_trace();
+#endif
       help_text.append("\nSearch for `")
           .append(cudaGetErrorName(error))
 #if defined(USE_ROCM)
@@ -85,6 +106,9 @@ const char* get_cuda_blocking_message() noexcept {
 
 // NOLINTNEXTLINE(bugprone-exception-escape,-warnings-as-errors)
 const char* get_cuda_check_suffix() noexcept {
+#if defined(USE_ROCM)
+  print_stack_trace();
+#endif
   return get_cuda_blocking_message();
 }
 
-- 
2.43.0

